internal/pkg/plugin/devlakeconfig/devlakeconfig.go

```diff
@@ -55,42 +58,118 @@ func ApplyConfig(options plugininstaller.RawOptions) error {
 func createConnections(host string, pluginName string, connections []Connection) error {
 	for i, c := range connections {
 		log.Infof("Connection %d: %s", i, c.Name)
-		configs, err := json.Marshal(c)
-		if err != nil {
+		if err := createConnection(host, pluginName, c); err != nil {
 			return err
 		}
-		log.Debugf("Connection configs: %s", string(configs))
-
-		url := fmt.Sprintf("%s/plugins/%s/connections", strings.TrimRight(host, "/"), pluginName)
-		log.Debugf("URL: %s", url)
+	}
+	log.Infof("All %s connections have been created.", pluginName)
+	return nil
+}
 
-		if err := createConnection(url, configs); err != nil {
-			return err
-		}
+func createConnection(host string, pluginName string, c Connection) error {
+	configs, err := json.Marshal(c)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(configs))
+	log.Debugf("Connection configs: %s", string(configs))
+	url := fmt.Sprintf("%s/plugins/%s/connections", strings.TrimRight(host, "/"), pluginName)
+	log.Debugf("URL: %s", url)
+	if _, err = apiRequest(http.MethodPost, url, configs); err != nil {
+		return err
 	}
+	return nil
+}
 
-	log.Infof("All %s connections have been created.", pluginName)
+func updateConnection(host string, pluginName string, c Connection) error {
+	id, err := getConnection(host, pluginName, c)
+	if err != nil {
+		return err
+	}
+	if id == 0 {
+		return nil
+	}
+	configs, err := json.Marshal(c)
+	if err != nil {
+		return err
+	}
+	log.Debugf("UPDATE Connection configs: %s", string(configs))
+	url := fmt.Sprintf("%s/plugins/%s/connections/%d", strings.TrimRight(host, "/"), pluginName, id)
+	log.Debugf("URL: %s", url)
+	if _, err = apiRequest(http.MethodPatch, url, configs); err != nil {
+		return err
+	}
 	return nil
 }
 
-func createConnection(url string, bodyWithJson []byte) error {
-	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(bodyWithJson))
+func deleteConnection(host string, pluginName string, c Connection) error {
```

Comment:
How about to add some blank lines or comments in this function? I think it will be more readable.